be/src/storage/lake/compaction_task_context.cpp

```diff
@@ -0,0 +1,43 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "storage/lake/compaction_task_context.h"
+
+#include <rapidjson/document.h>
+#include <rapidjson/rapidjson.h>
+#include <rapidjson/stringbuffer.h>
+#include <rapidjson/writer.h>
+
+namespace starrocks::lake {
+
+std::string CompactionTaskContext::to_json_stats() {
+    rapidjson::Document root;
+    root.SetObject();
+    auto& allocator = root.GetAllocator();
+    // add stats
+    root.AddMember("reader_total_time_ms", rapidjson::Value(stats->reader_time_ns / 1000000), allocator);
+    root.AddMember("reader_io_ms", rapidjson::Value(stats->io_ns / 1000000), allocator);
+    root.AddMember("reader_io_count_remote", rapidjson::Value(stats->io_count_remote), allocator);
+    root.AddMember("reader_io_count_local_disk", rapidjson::Value(stats->io_count_local_disk), allocator);
+    root.AddMember("compressed_bytes_read", rapidjson::Value(stats->compressed_bytes_read), allocator);
+    root.AddMember("segment_init_ms", rapidjson::Value(stats->segment_init_ns / 1000000), allocator);
+    root.AddMember("column_iterator_init_ms", rapidjson::Value(stats->column_iterator_init_ns / 1000000), allocator);
+    root.AddMember("segment_write_ms", rapidjson::Value(stats->segment_write_ns / 1000000), allocator);
```

Comment:
I just realized that while you are reading these variables of type `int64_t` in this thread, another thread may be modifying it, and this is undefined behavior. Maybe we should check if `finish_time` is greater than 0 (finish_time greater than 0 means that the compaction thread won't be anymore) before reading `context->stats`, like we do with `context->status`, or maybe we should replace `int64_t` with `std::atomic<int64_t>`?

Reply:
Thanks, I will use `std::atomic<int64_t>` instead.

> I just realized that while you are reading these variables of type `int64_t` in this thread, another thread may be modifying it, and this is undefined behavior. Maybe we should check if `finish_time` is greater than 0 (finish_time greater than 0 means that the compaction thread won't modify context anymore) before reading `context->stats`, like we do with `context->status`, or maybe we should use `std::atomic<int64_t>`?

Reply:
> I just realized that while you are reading these variables of type `int64_t` in this thread, another thread may be modifying it, and this is undefined behavior. Maybe we should check if `finish_time` is greater than 0 (finish_time greater than 0 means that the compaction thread won't modify context anymore) before reading `context->stats`, like we do with `context->status`, or maybe we should use `std::atomic<int64_t>`?

`stats` need to be updated and watched at running time, so i think we can't use `finish_time` checking logic as `context->status`